train_prototree.py

```diff
@@ -13,9 +15,11 @@
 from prototree.visualization import save_prototype_visualizations
 from util.args import get_args, get_optimizer
 from util.data import get_dataloaders
-from util.log import Log
 from util.net import BASE_ARCHITECTURE_TO_FEATURES
 
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
```

Comment:
the `__name__` is not the best choice in scripts I find, it will always be "main" and is thus not very informative. You could use `getLogger("train_prototree")` or get the name of the file from `__file__` instead

Reply:
Good idea, I've changed this now.